R/label-placer.R

```diff
@@ -201,6 +203,46 @@ label_placer_manual <- function(breaks, x, y, theta) {
 }
 
 
+#' @param rot_adjuster Function that standardizes the rotation angles of the labels.
+#'   See e.g. [`angle_halfcircle_bottom()`].
+#' @rdname label_placer
+#' @export
+label_placer_middle <- function(rot_adjuster = angle_halfcircle_bottom(), ...) {
+  placer_fun <- function(line_data, ...) {
+    out <- data.frame(x = numeric(0), y = numeric(0), theta = numeric(0))
+
+    # It identifies each isoline subdivision. For an individual isoline the id column identifies the number of subdivisions.
+    line_sections <- unique(line_data$id)
+
+    # Then the label is printed at the middle of each isoline subdivision.
+    for (i in 1:length(line_sections)) {
+      x <- line_data$x[line_data$id == i]
+      y <- line_data$y[line_data$id == i]
+
+      middle_index <- as.integer(length(x) / 2)
+
+      x_ <- x[middle_index]
+      y_ <- y[middle_index]
```

Comment:
I don't particularly like `x_` and `y_`. How about `x_mid` and `y_mid` or similar?

Reply:
Ok. I will rename these two variables.